src/lib/decoders.ts

```diff
@@ -0,0 +1,118 @@
+import { Result } from '@ethersproject/abi';
+import BigNumber from 'bignumber.js';
+import { BytesLike, defaultAbiCoder } from 'ethers/lib/utils';
+import { parseInt } from 'lodash';
+import { BN_0 } from '../bignumber-constants';
+import { NULL_ADDRESS } from '../constants';
+import { MultiResult } from './multi-wrapper';
+
+export function generalDecoder<T>(
+  result: MultiResult<BytesLike>,
+  types: string[],
+  defaultValue: T,
+  parser?: (v: Result) => T,
+): T {
+  if (!result.success || result.returnData === '0x') {
+    return defaultValue;
+  }
+  const decoded = defaultAbiCoder.decode(types, result.returnData);
+  return parser ? parser(decoded) : decoded[0];
+}
+
+export const uintDecode = (result: MultiResult<string>): bigint => {
+  if (!result.success) {
+    return 0n;
+  }
+  return BigInt(defaultAbiCoder.decode(['uint'], result.returnData)[0]);
+};
+
+export const uint256ArrayDecode = (result: MultiResult<string>): bigint => {
+  if (!result.success) {
+    return 0n;
+  }
+  return defaultAbiCoder
+    .decode(['uint256[]'], result.returnData)[0]
+    .map((r: BigNumber) => BigInt(r.toString()));
+};
+
+export const uin256DecodeToBigNumber = (
+  result: MultiResult<string>,
+): BigNumber => {
+  if (!result.success) {
+    return BN_0;
+  }
+  return new BigNumber(
+    defaultAbiCoder.decode(['uint256'], result.returnData)[0],
+  );
+};
```

Comment:
This function doesn't work (you get NaN from `new BigNumber(ethers.utils.defaultAbiCoder.decode(['uint'], '0x0000000000000000000000000000000000000000000000000000000000000011')[0]).toString()`).  You can make it work by converting to string before BigNumber.

Reply:
Added `toString()`